File: plugin/forem-nvim.lua
local M = {}
local forem = require("forem-nvim")
local notify = require("forem-nvim.notify")

local commands = {
  feed = "feed",
  my_articles = "my_articles",
  new_article = "new_article",
  open_url = "open_url"
}

vim.api.nvim_create_user_command(
  "Forem",
  function(data)
    local args = data.args

    if args == commands.feed then
      return forem.feed()
    elseif args == commands.my_articles then
      return forem.my_articles()
    elseif args == commands.new_article then
      return forem.new_article()
    elseif args == commands.open_url then
      return forem.open_url()
    else
      notify.error("Unknown command: " .. args)
    end
  end,
  {
    nargs = 1,
    complete = function() return vim.tbl_values(commands) end
  }
)
return M

File: plugin/forem-nvim.ts
import * as forem from "forem-nvim";
import * as notify from "forem-nvim.notify";

enum Command {
  Feed = "feed",
  MyArticles = "my_articles",
  NewArticle = "new_article",
  OpenUrl = "open_url",
}

const isCommand = (command: string): command is Command =>
  Object.values(Command).includes(command as Command);

vim.api.nvim_create_user_command(
  "Forem",
  function ({ args }) {
    if (!isCommand(args)) {
      notify.error(`Unknown command: ${args}`);
      return;
    }

    switch (args) {
      case Command.Feed:
        return forem.feed();
      case Command.MyArticles:
        return forem.my_articles();
      case Command.NewArticle:
        return forem.new_article();
      case Command.OpenUrl:
        return forem.open_url();
    }
  },
  { nargs: 1, complete: () => Object.values(Command) },
);
